__tests__/app.test.js

```diff
@@ -41,3 +41,31 @@ describe("GET /api/topics", () => {
       });
   });
 });
+
+describe("GET /api/articles/:article_id", () => {
+  test("returns status 200", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 1,
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          votes: 100,
+        });
+      });
+  });
+
+  test("returns status 404 with non-existant article_id", () => {
+    return request(app)
+      .get("/api/articles/199")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No article found for article_id: 199");
+      });
+  });
+});
```

Comment:
nice error testing here. Just 1 more to consider, what happens if there is a bad request with an invalid article id (incorrect data type)? Think PSQL errors/400

Reply:
added error handling for 22P02 PSQL error and additional test